# Bonus: Temperature Analysis I

In [8]:
!pip install fsspec

In [1]:
import sqlalchemy
from sqlalchemy.ext.automap import automap_base
from sqlalchemy.orm import Session
from sqlalchemy import create_engine, func
import pandas as pd
import numpy as np
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] =pd.to_datetime(df['date'], format='%Y-%m-%d')
df.dtypes

station            object
date       datetime64[ns]
prcp              float64
tobs                int64
dtype: object

In [4]:
# Set the date column as the DataFrame index
new_df=df.set_index('date', drop=True)
new_df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

In [ ]:
# Drop the date column

### Compare June and December data across all years 

In [9]:
from scipy import stats

In [11]:
# Filter data for desired months
# Filter data for desired months
june_df=new_df[new_df.index.month==6].dropna()
december_df=new_df[new_df.index.month==12].dropna()
print(june_df,december_df)

                station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77
...                 ...   ...   ...
2017-06-26  USC00516128  0.02    79
2017-06-27  USC00516128  0.10    74
2017-06-28  USC00516128  0.02    74
2017-06-29  USC00516128  0.04    76
2017-06-30  USC00516128  0.20    75

[1574 rows x 3 columns]                 station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64
...                 ...   ...   ...
2016-12-27  USC00516128  0.14    71
2016-12-28  USC00516128  0.14    71
2016-12-29  USC00516128  1.03    69
2016-12-30  USC00516128  2.37    65
2016-12-31  USC00516128  0.90    65

[1405 rows x 3 columns]


In [14]:
# Identify the average temperature for June

avg_june =round(june_df['tobs'].mean())
print(f'The average temperature in Hawaii for the month of June across all years is {avg_june}F.')

The average temperature in Hawaii for the month of June across all years is 75.0F.


In [13]:
# Identify the average temperature for December

avg_december =round(december_df['tobs'].mean())
print(f'The average temperature in Hawaii for the month of December across all years is {avg_december}F.')

The average temperature in Hawaii for the month of December across all years is 71.0F.


In [15]:
# Create collections of June temperature data
june_df['tobs'].describe()

count    1574.000000
mean       74.887548
std         3.255695
min        65.000000
25%        72.000000
50%        75.000000
75%        77.000000
max        85.000000
Name: tobs, dtype: float64

In [16]:
# Create collections of December temperature data
december_df['tobs'].describe()

count    1405.000000
mean       70.930249
std         3.741578
min        56.000000
25%        69.000000
50%        71.000000
75%        74.000000
max        81.000000
Name: tobs, dtype: float64

In [17]:
# Run paired t-test
stats.ttest_ind(june_df['tobs'], december_df['tobs'], equal_var = False)

Ttest_indResult(statistic=30.624201480767336, pvalue=6.622829250184814e-178)

### Analysis

The temperatures in June and December have a difference statistically. With a p-value at 6.6 it indicates that a large difference exists between the two sample sets. 
